frog/imports/ui/engine/ActivityDisplay.jsx

```diff
@@ -0,0 +1 @@
+﻿import React, { Component, PropTypes } from 'react';
import ReactDOM from 'react-dom';
import ReactPlayer from 'react-player';
import QuestionDisplay from './QuestionDisplay.jsx';

export default class ActivityDisplay extends Component {

  constructor(props) {
    super(props);

    this.state = {
      isClicked: false,
    }
  }

  activityHandler(event) {
    event.preventDefault();
    this.setState({
      isClicked: !this.state.isClicked,
    });
  }

  renderQuestions(){
    return (this.props.object.questions ?
      this.props.object.questions
        .map((questionz) => 
          (<QuestionDisplay 
            key = {String(this.props._id).concat(questionz.question)}
            id = {String(this.props._id).concat(questionz.question)}
            question = {questionz.question}
            choices = {questionz.choices}
            answer = {questionz.answer} />
          )) : <li>No Questions</li>);
  }

  renderActivityOnClick(){
    if (this.props.type == "lecture"){
      return (<div className="display-lecture">
                  Follow the URL: {this.props.object.path}
              </div>);
    } else if (this.props.type == "video"){
      return (<ReactPlayer url={this.props.object.url} playing controls/>);
    } else if (this.props.type == "quiz"){
      return (<div className="display-quiz">
                      {this.renderQuestions()}
              </div>);
    } else {
      return "";
    };
  }

  render() {
    return (
      <div className="activity-summary">
      <li onClick={this.activityHandler.bind(this)}>
        {this.props.name}, type:{this.props.type}, plane:{this.props.plane}
      </li>

      {
        //If the user has clicked on the activity, put prevous properties in addition
        //to hidden properties (returned by the renderObjectProperties)
        this.state.isClicked ? this.renderActivityOnClick() : ""
      }

      </div>
    );
  }

}

```

Comment:
I don't know why I can't make this comment bellow on this file... 

line 27-28
```javascript
renderQuestions(){
     return (this.props.object.questions ?
       this.props.object.questions
         .map((questionz) => 
           (<QuestionDisplay 
             key = {String(this.props._id).concat(questionz.question)}
             id = {String(this.props._id).concat(questionz.question)}
             question = {questionz.question}
             choices = {questionz.choices}
             answer = {questionz.answer} />
           )) : <li>No Questions</li>);
   }
```
Get the folowing error
```
modules.js?hash=71f89b4…:21433 Warning: flattenChildren(...): Encountered two children with the same key, `undefinedq`. Child keys must be unique; when two children share a key, only the first child will be used.
    in div (created by ActivityDisplay)
    in div (created by ActivityDisplay)
    in ActivityDisplay (created by Engine)
    in ul (created by Engine)
    in div (created by Engine)
    in Engine (created by MeteorDataContainer)
    in MeteorDataContainer (created by Body)
    in Body (created by MeteorDataContainer)
    in MeteorDataContainer (created by App)
    in div (created by App)
    in div (created by App)
    in App
```
The key is not well chosen. If two question are the same only one will be displayed. It can be the case, for example if the question is "Chose the right definition from the definitions bellow". You should try to map with the index of the array and use `key = {String(this.props._id).concat(***index***)}`